optica.rb

```diff
@@ -8,6 +8,15 @@ class Optica < Sinatra::Base
   end
 
   get '/' do
+    return get_nodes(request, true)
+  end
+
+  # endpoint for fab usage
+  get '/roles' do
+    return get_nodes(request, false)
+  end
+
+  def get_nodes(request, is_full_node)
```

Comment:
suggestion: maybe make `is_full_node` something like `fields_to_include`. if it's empty, do no `slice`ing. if it has some fields, then `slice(*fields_to_include)`. that would support future use cases.